lxd/storage/drivers/driver_pure_volumes.go

```diff
@@ -566,3 +697,13 @@ func (d *pure) RenameVolumeSnapshot(snapVol Volume, newSnapshotName string, op *
 	// Renaming a volume snapshot won't change an actual name of the Pure Storage volume snapshot.
 	return nil
 }
+
+// getSnapshotParentVolume returns a parent volume that has volatile.uuid set to the snapshot's parent UUID.
+func getSnapshotParentVolume(snapVol Volume) Volume {
```

Comment:
lets move this to generic utils rather than in the pool driver file, as although only the pure driver uses this, it is not pure specific logic.

Reply:
or snapVol.GetParentVolume()?